rust/lancedb/src/remote/client.rs

```diff
@@ -21,13 +21,65 @@ use reqwest::{
 
 use crate::error::{Error, Result};
 
+// We use the `HttpSend` trait to abstract over the `reqwest::Client` so that
+// we can mock responses in tests. Based on the patterns from this blog post:
+// https://write.as/balrogboogie/testing-reqwest-based-clients
 #[derive(Clone, Debug)]
-pub struct RestfulLanceDbClient {
+pub struct RestfulLanceDbClient<S: HttpSend = Sender> {
```

Comment:
Overly opinionated personal comment (not even rising to the standards of minor nit): I feel like parameterizing this (instead of using `Box<dyn HttpSend>`) is probably overkill.